# Allman style braces

Checks for the allman style (braces should be on their own line)

## Noncompliant Code Example

```d
if (param < 0) {
}
```

## Compliant Solution

```d
if (param < 0)
{
}
```

See [Allman style](https://en.wikipedia.org/wiki/Indentation_style#Allman_style).
